proposals/NNNN-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,239 @@
+
+The `let` declared property can also be initialized once via the generated storage property. For example, reusing the above example:
+```swift
+  @BoilingPoint let water: Double
+  
+  init() {
+    _water = BoilingPoint(wrappedValue: 373.1)
+  }
+```
+
+After the `let` declared property has been initialized, attempting to reassign it will show an error. Clients can continue to manipulate the projection just like with `var` declared properties.
+```swift
+  temperature.water = 400.0 // error: cannot assign to property: 'water' is a 'let' constant
+  temperature.$water.toFarenheit() // converts to 212.0
+```
+
+Property wrappers with `let` declarations will be allowed both as members and local declarations, as envisioned by [SE-0258](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md) for `var` declared property wrappers. All other property wrapper traits also remain unchanged from SE-0258.
+
+## Detailed Design
+
+Marking a property wrapped declaration with a `let` makes that property a getter-only computed property and introduces a `let` synthesized stored property whose type is the wrapper type. This expands where and how property wrappers can be used in Swift.
+
+### Nonisolated property wrappers
+
+Actor methods and properties, which are inherently isolated to that actor, can be marked `nonisolated` to allow them to be accessed from outside their actor context. At this time, the `nonisolated` keyword cannot be used on property wrappers because `var` declared property wrappers generates a `var` declared backing property that is not safe from race conditions and so cannot be `nonisolated`.
+```swift
+@propertyWrapper
+struct Wrapper {
+  var wrappedValue: Int { .zero }
+  var projectedValue: Int { .max }
+}
+
+@MainActor
+class C {
+  @Wrapper nonisolated var value // error: `nonisolated` is not supported on `var` declared property wrappers
+}
+```
+
+Let declared property wrappers will now allow us to write nonisolated property wrappers as the backing property will also be stored `let` property. 
+```swift
+@MainActor
+class C {
+  @Wrapper nonisolated let value: Int
+  
+  nonisolated func test() {
+    _ = value
+    _ = $value
+  }
+}
+```
+
+### Property wrappers with backing reference types
+
+A `let` wrapped property could be useful for reference types like a property wrapper class. Typically property wrappers are written for value types but occasionally a protocol like `NSObject` may require the use of a class. For example:
+```swift
+@propertyWrapper
+class WrapperClass<T> : NSObject {
+  var wrappedValue: T
+
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+}
+
+class C {
+  @WrapperClass let value: Int
+
+  init(v: Int) {
+    value = v
+  }
+}
+```
+Even though  `value` is declared with a class type property wrapper, it can only be assigned once, preventing any future unintentional changes to the property wrapper class instance in this context. 
```

Comment:
```suggestion

Even though  `value` is declared with a class type property wrapper, it can only be assigned once, preventing any future unintentional changes to the property wrapper class instance in this context. 
```

Presumably, this is because the synthesized computed property is getter-only, is that correct? If so, it would be helpful to state it.